plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -1031,6 +1036,72 @@ public void onClose(KubernetesClientException cause) {
   }
 
   @Override
+  public ContainerProcesses top(final TopParams params) throws IOException {
+    String containerId = params.getContainer();
+    Pod pod = getChePodByContainerId(containerId);
+    String podName = pod.getMetadata().getName();
+    String[] command;
+    final String[] psArgs = params.getPsArgs();
+    if (psArgs != null && psArgs.length != 0) {
+      int length = psArgs.length + 1;
+      command = new String[length];
+      command[0] = PS_COMMAND;
+      System.arraycopy(psArgs, 0, command, 1, psArgs.length);
+    } else {
+      command = new String[1];
+      command[0] = PS_COMMAND;
+    }
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    ContainerProcesses processes = new ContainerProcesses();
+    OpenShiftClient openShiftClient = new DefaultOpenShiftClient();
+    try (ExecWatch watch =
+        openShiftClient
+            .pods()
+            .inNamespace(openShiftCheProjectName)
+            .withName(podName)
+            .redirectingOutput()
+            .redirectingError()
+            .exec(command)) {
+      Thread.sleep(2500);
```

Comment:
is it mandatory this wait ?

Reply:
I asked @snjeza and was told that it was copied from the exec method. I believe it was copied from this https://github.com/JPinkney/che/blob/0c86b9b4c25e045e657a1dbca850fdb1c3c78e5d/plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java#L1136-L1155 as far as I can see the ExecutorService declared isn't even used except for shutting down in the finally block so this line wouldn't be needed.